### DISCUSSION TITLE
coordinates for SubquoModuleElem

### DISCUSSION THREAD

**Discussion by JohnAAbbott** [oscar-system/Oscar.jl#2346] [General]
I had expected `coordinates` to work the same on a `SubquoModuleElem` as on `FreeMod` elem; instead it gives an unexpected result.  Are my expectations incorrect?
```
S, (x, y, z) = graded_polynomial_ring(QQ, ["x", "y", "z"], [1 0 1; 0 1 1]); # hard to read

GG = grading_group(S); ### apparently not the same as free_abelian_group(2);
FM_grad = graded_free_module(S, GG.([[1, 2], [-1, 3]]));
(ee1,ee2) = gens(FM_grad);

SS,_ = sub(FM_grad, [(x^2+y^2)*ee1 + (x*y+y*z+z*x)*ee2,  (x*y+y*z+z*x)*ee1 + (x^2+y^2)*ee2] );

LM = leading_module(SS); 
GLM = gens(LM);
GLM[1]
coordinates(GLM[1])    # unexpected result
coordinates(x^2*ee1)  # what I expected
```

**Comment by JohnAAbbott** [comment]
One solution/work-around would be to "lift" the `SubquoModuleElem` to a `FreeModElem`.  But how?
Since there is no quotient here, lifting could be automatic.... or not?

**Comment by wdecker** [comment]
But everything is correct. You are confusing what the coordinates are.
Just check the docu in both cases. you may also enter coordinates(x^2*GLM[1])

**Comment by JohnAAbbott** [comment]
Ouch!  That was a subtlety I'd overlooked.

**Comment by JohnAAbbott** [comment]
I'd like to suggest that the manual make it clear that an element of a
submodule is not considered (by Oscar) to be an element of the ambient
free module; the change of point of view is achieved by the function
`ambient_representative`.

I wonder whether the function/method `coordinates` which operates on an
element of a submodule should have a name different from `coordinates`
which operates on an element of a free module.  Personally I had
expected `coordinates` always to give the representation in terms of the
generators of the free module -- right now I cannot even find the manual
entries for `coordinates` as applied to elements of (sub-)modules.

Here is an example:
```
S, (x, y, z) = polynomial_ring(QQ, ["x", "y", "z"]);
FM = free_module(S, 2);
(ee1,ee2) = gens(FM);

SubM,_ = sub(FM, [(x^2+y^2)*ee1 + (x*y+y*z+z*x)*ee2,
                  (x*y+y*z+z*x)*ee1 + (x^2+y^2)*ee2] );

LM = leading_module(SubM);

GLM = gens(LM);
GLM[1]
v = x^2*ee1  # looks the same, but is not!
coordinates(GLM[1])    # different results
coordinates(x^2*ee1)   # different results
```

**Reply by JohnAAbbott** [reply]
Let me clarify: I cannot find the manual description of `coordinates` in the **web-version** of the Oscar manual.  In a Julia/Oscar session the `?` operator does find a description (among many others).
